test/e2eTests/extension.e2e.test.ts

```diff
@@ -0,0 +1,84 @@
+import vscode, { Uri, Diagnostic, Range } from 'vscode';
+import assert from 'assert';
+import path from 'path';
+import { ensureExtensionActivatedAndInitializationFinished } from '../utils/testHelper';
+import { EventType } from '../../src/common/eventType';
+import { BaseEvent, UserSignInCompleted, BuildCompleted } from '../../src/common/loggingEvents';
+import { createSandbox, SinonSandbox } from 'sinon';
+import { uriHandler } from '../../src/shared';
+import { DocfxExecutionResult } from '../../src/build/buildResult';
+
+describe('E2E Test', () => {
+    let sinon: SinonSandbox;
+    before(() => {
+        if (!process.env.VSCODE_DOCS_BUILD_EXTENSION_BUILD_USER_TOKEN) {
+            throw new Error('Cannot get "VSCODE_DOCS_BUILD_EXTENSION_BUILD_USER_TOKEN" from environment variable');
+        }
+
+        const fakedGitHubCallbackURL = <vscode.Uri>{
+            authority: 'ceapex.docs-build',
+            path: '/github-authenticate',
+            query: `id=vsc-service-account-id&name=VSC-Service-Account&email=vscavu@microsoft.com&X-OP-BuildUserToken=${process.env.VSCODE_DOCS_BUILD_EXTENSION_BUILD_USER_TOKEN}`
+        };
+
+        sinon = createSandbox();
+        sinon.stub(vscode.env, 'openExternal').callsFake(
+            function (target: vscode.Uri): Thenable<boolean> {
+                return new Promise((resolve, reject) => {
+                    setTimeout(() => {
+                        uriHandler.handleUri(fakedGitHubCallbackURL);
+                    }, 10);
+                    resolve(true);
+                });
+            }
+        );
+    });
+
+    after(() => {
+        sinon.restore();
+    });
+
+    it('Sign in to Docs and trigger build', (done) => {
+        (async function () {
+            const extension = await ensureExtensionActivatedAndInitializationFinished();
+            assert.notEqual(extension, undefined);
+
+            extension.exports.eventStream.subscribe((event: BaseEvent) => {
+                switch (event.type) {
+                    case EventType.UserSignInCompleted:
+                        let asUserSignInCompleted = <UserSignInCompleted>event;
+                        assert.equal(asUserSignInCompleted.succeeded, true);
+                        triggerBuild();
+                        break;
+                    case EventType.BuildCompleted:
+                        finalCheck(<BuildCompleted>event);
+                        done();
+                        break;
+                }
+            });
+
+            triggerSignIn();
+
+            function triggerSignIn() {

```

Comment:
does this need to be a function? just 1 line call
And also, there is another method
https://github.com/docascode/vscode-docs-build/blob/1a6faa84e966f462fc7e0eb5826cf03b0ba94700/test/utils/testHelper.ts#L20